arch/xtensa/src/esp32s2/esp32s2_tim.c

```diff
@@ -0,0 +1,981 @@
+      modifyreg32(TIMG_INT_CLR_TIMERS_REG(priv->gid), 0, TIMG_T0_INT_CLR_M);
+    }
+  else
+    {
+      modifyreg32(TIMG_INT_CLR_TIMERS_REG(priv->gid), 0, TIMG_T1_INT_CLR_M);
+    }
+}
+
+/****************************************************************************
+ * Name: esp32s2_tim_checkint
+ *
+ * Description:
+ *   Check the interrupt status bit.
+ *
+ * Parameters:
+ *   dev           - Pointer to the timer driver struct.
+ *
+ * Returned Values:
+ *  Return 1 in case of an interrupt is triggered, otherwise 0.
+ *
+ ****************************************************************************/
+
+static int esp32s2_tim_checkint(FAR struct esp32s2_tim_dev_s *dev)
+{
+  struct esp32s2_tim_priv_s *priv = (struct esp32s2_tim_priv_s *)dev;
+  uint32_t reg_value;
+  int ret;
+
+  DEBUGASSERT(dev);
+
+  if (priv->tid == TIMER0)
+    {
+      reg_value = getreg32(TIMG_INT_ST_TIMERS_REG(priv->gid));
+      ret = REG_MASK(reg_value, TIMG_T0_INT_ST);
+    }
+  else
+    {
+      reg_value = getreg32(TIMG_INT_ST_TIMERS_REG(priv->gid));
+      ret = REG_MASK(reg_value, TIMG_T1_INT_ST);
+    }
+
+  return ret;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: esp32s2_tim_init
+ *
+ * Description:
+ *   Initialize TIMER device.
+ *
+ * Parameters:
+ *   timer           - Timer instance to be initialized.
+ *                     Valid values: 0 or 1.
+ *
+ * Returned Values:
+ *   If the initialization is successful, return a pointer to the timer
+ *   driver struct associated to that timer instance.
+ *   In case it fails, return NULL.
+ *
+ ****************************************************************************/
+
+FAR struct esp32s2_tim_dev_s *esp32s2_tim_init(int timer)
+{
+  FAR struct esp32s2_tim_priv_s *tim = NULL;
+
+  /* First, take the data structure associated with the timer instance */
+
+  switch (timer)
+    {
+#ifdef CONFIG_ESP32S2_TIMER0
+      case 0:
+        {
+          tim = &g_esp32s2_tim0_priv;
+          break;
+        }
+#endif
+
+#ifdef CONFIG_ESP32S2_TIMER1
+      case 1:
+        {
+          tim = &g_esp32s2_tim1_priv;
+          break;
+        }
+#endif
+
+#ifdef CONFIG_ESP32S2_TIMER2
+      case 2:
+        {
+          tim = &g_esp32s2_tim2_priv;
+          break;
+        }
+#endif
+
+#ifdef CONFIG_ESP32S2_TIMER3
+      case 3:
+        {
+          tim = &g_esp32s2_tim3_priv;
+          break;
+        }
+#endif
+
+      default:
+        {
+          tmrerr("ERROR: unsupported TIMER %d\n", timer);
+          goto errout;
+        }
+    }
+
+  /* Verify if it is in use */
+
+  if (tim->inuse == false)
+    {
+      tim->inuse = true;  /* If it was not, now it is */
+    }
+  else
+    {
+      tmrerr("ERROR: TIMER %d is already in use\n", timer);
+      tim = NULL;
+    }
+
+  errout:
```

Comment:
```suggestion
errout:
```
Fix indentation. According to the coding standard:
> Labels are never indented. Labels must always begin in column 1.

Reply:
Done